test/data/ansible/check_ostree.yaml

```diff
@@ -337,11 +337,33 @@
         - name: failed count + 1
           set_fact:
             failed_counter: "{{ failed_counter | int + 1 }}"
-      when: ansible_facts['distribution'] != 'RedHat' and ansible_facts ['distribution_version'] != '8.5'
+      when: ansible_facts['distribution'] != 'RedHat' and ansible_facts ['distribution_version'] is version('8.5', '!=')
 
     # case: check installed greenboot packages
     # https://github.com/osbuild/osbuild-composer/blob/master/internal/distro/rhel8/distro.go#L634
-    - name: greenboot and it's related packages should be installed
+    - name: greenboot and it's related packages should be installed (RHEL 8.6+ and CS9)
+      block:
+        - name: greenboot and it's related packages should be installed
+          shell: rpm -qa | grep greenboot
+          register: result_greenboot_packages
+
+        - assert:
+            that:
+              - "'greenboot-0' in result_greenboot_packages.stdout"
+              - "'greenboot-default-health-checks' in result_greenboot_packages.stdout"
```

Comment:
Similar to https://github.com/osbuild/osbuild-composer/pull/2196#discussion_r806057954. Same comments apply. @nullr0ute if you know a better way to do this let's open a ticket for it but not delay this PR on it.

Reply:
Issue opened https://github.com/osbuild/osbuild-composer/issues/2302